component/otelcol/config_grpc.go

```diff
@@ -136,11 +137,11 @@ type GRPCClientArguments struct {
 	TLS       TLSClientArguments        `river:"tls,block,optional"`
 	Keepalive *KeepaliveClientArguments `river:"keepalive,block,optional"`
 
-	ReadBufferSize  units.Base2Bytes  `river:"read_buffer_size,attr,optional"`
-	WriteBufferSize units.Base2Bytes  `river:"write_buffer_size,attr,optional"`
-	WaitForReady    bool              `river:"wait_for_ready,attr,optional"`
-	Headers         map[string]string `river:"headers,attr,optional"`
-	BalancerName    string            `river:"balancer_name,attr,optional"`
+	ReadBufferSize  units.Base2Bytes               `river:"read_buffer_size,attr,optional"`
+	WriteBufferSize units.Base2Bytes               `river:"write_buffer_size,attr,optional"`
+	WaitForReady    bool                           `river:"wait_for_ready,attr,optional"`
+	Headers         map[string]configopaque.String `river:"headers,attr,optional"`
```

Comment:
Thanks, I agree! I do think that we should probably use River secrets for this, but I raised a separate issue for it just now (#4185) so that we can tackle this separately. For the purpose of this PR I agree we should limit the impact to just an Otel upgrade without changing previous behaviour.